src/software/proto/primitive/primitive_msg_factory.cpp

```diff
@@ -62,11 +62,40 @@ std::unique_ptr<TbotsProto::PrimitiveNew> createMovePrimitive(
     return move_primitive_msg;
 }
 
-std::unique_ptr<TbotsProto::PrimitiveNew> createSpinningMovePrimitive(
+std::unique_ptr<TbotsProto::Primitive> createLegacyMovePrimitive(
+    const Point &dest, const Angle &final_angle, double final_speed,
+    DribblerEnable enable_dribbler, MoveType move_type, AutochickType autochick)
+{
+    // TODO (#1638): Remove this and use createMovePrimitive, createAutochipMovePrimitive,
+    // or createAutokickMovePrimitive instead. This should be done after trajectory
+    // planner is integrated in case the arguments for creating createMovePrimitive,
+    // createAutochipMovePrimitive, and createAutokickMovePrimitive changes.
+    if (autochick == AutochickType::AUTOKICK)
+    {
+        return createAutokickMovePrimitive(
+            dest, final_speed, move_type == MoveType::SLOW, final_angle,
+            enable_dribbler == DribblerEnable::ON ? 16000 : 0,
+            BALL_MAX_SPEED_METERS_PER_SECOND - 1);
+    }
+    else if (autochick == AutochickType::AUTOCHIP)
+    {
+        return createAutochipMovePrimitive(
+            dest, final_speed, move_type == MoveType::SLOW, final_angle,
+            enable_dribbler == DribblerEnable::ON ? 16000 : 0, 2);
```

Comment:
Why is the `chip_distance_meters` hard coded to 2 here? Is this what the old `AUTOCHIP` used to do?